lib/beaker/options/parser.rb

```diff
@@ -228,10 +228,22 @@ def parse_hosts_options
         hosts_options = Beaker::Options::HostsFileParser.parse_hosts_file(@options[:hosts_file])
         return hosts_options
       rescue Errno::ENOENT
-        require 'tempfile'
+        error_message = "\nHosts file '#{@options[:hosts_file]}' does not exist."
+        error_message << "\nTrying as beaker-hostgenerator input.\n\n"
+        $stderr.puts error_message
```

Comment:
If you think about the feature we're introducing and not how the code is written, it's not correct to say that using BHG is an error.  It's not.  The feature we're introducing is the ability to have `--hosts` be a file or a BHG string, so why would we print a stdout message for one and a stderr message for the other?  

They're both equally valid and formally supported features, so they should just be logged like normal behavior; not an error.

Regarding the implementation, why wouldn't we do a `File.exists(@options[:hosts_file])` first to determine which code path we're going down?  That would be better than using exception for normal flow-of-control.

If we're forced to keep it wrapped in a try/catch like this, I still don't see why we would print to stderr yet.  It makes sense to print to stderr if the subsequent call to BHG fails.